src/Simplex/Chat/Stats.hs

```diff
@@ -0,0 +1,86 @@
+{-# LANGUAGE DuplicateRecordFields #-}
+{-# LANGUAGE TemplateHaskell #-}
+
+module Simplex.Chat.Stats where
+
+import qualified Data.Aeson.TH as J
+import Simplex.Chat.Types
+import Simplex.Messaging.Agent.Client
+import Simplex.Messaging.Agent.Stats
+import Simplex.Messaging.Parsers (defaultJSON)
+import Simplex.Messaging.Protocol
+
+data PresentedServersSummary = PresentedServersSummary
+  { userServersSummary :: ServersSummary,
+    totalServersSummary :: ServersSummary
+  }
+  deriving (Show)
+
+data ServersSummary = ServersSummary
+  { currentlyUsedSMPServers :: [SMPServerSummary],
+    previouslyUsedSMPServers :: [SMPServerSummary],
+    onlyProxiedSMPServers :: [SMPServerSummary],
+    currentlyUsedXFTPServers :: [XFTPServerSummary],
+    previouslyUsedXFTPServers :: [XFTPServerSummary]
+  }
+  deriving (Show)
+
+data SMPServerSummary = SMPServerSummary
+  { smpServer :: SMPServer,
+    -- known:
+    -- always Nothing in totalServersSummary - to avoid hassle of navigating to other users server settings;
+    -- always Just in userServersSummary - True if server is in list of user servers, otherwise False;
+    -- True - allows to navigate to server settings, False - allows to add server to configured as known (SEKnown)
+    known :: Maybe Bool,
+    -- state:
+    -- Just if currently used in session, otherwise Nothing;
+    -- onlyProxiedSMPServers would always have state and statsData as Nothing
+    state :: Maybe SMPServerState,
+    -- statsData:
+    -- even if state is Nothing, statsData can be Just - server could be used earlier in session
+    -- or in previous sessions and stats for it were restored
+    statsData :: Maybe AgentSMPServerStatsData
+  }
```

Comment:
initially I tried more strict types that wouldn't allow impossible states, e.g. `CurrentlyUsedSMPServerSummary` had `state :: SMPServerState` (not Maybe), `PreviouslyUsedSMPServerSummary` didn't have `state`, `onlyProxiedSMPServers` was simply list `[SMPServer]`.

But after also accounting for `known` - it required creating yet another type for onlyProxiedSMPServers, because server can be disabled/known and still have no stats and state and so be "only proxied" - I got replaced these more precise types with a single `SMPServerSummary` with Maybe fields